### PR TITLE
Fix Should Exist assertion to handle paths with [ ]

### LINKED ISSUE #343: Special characters do not work with Should Exist
The first It works but the second does not. There is a need of ExistExactly.

```powershell
Describe "Testing Exists" {
    It "works with test-path" {
        $file = New-Item -Path "TestDrive:\[test].txt" -ItemType File         
        Test-Path -LiteralPath $file | Should Be $true
    }

    It "should work with Exists" {
        $file = New-Item -Path "TestDrive:\[test].txt" -ItemType File 
        $file | Should Exist        
    }
}
```

### PR DESCRIPTION
**Pull request by nohwnd**
Fixing bug which caused that paths that contain [ ] wildcard were not
handled correctly by the Should Exist assertion.

It's morning am not 100 % sure that this won't break anything. But my reasoning is that any path that is provided is expanded by PowerShell (like path with environment variables) so no special handling is needed there. Same goes for PS(Provider)Paths... nothing else comes to mind. 

Review please. :)

Fixes #343

### CODE REVIEW

**Comment by nohwnd** [comment]
So obviously I was thinking about files and registry only and forgot about function and env drives. Looking into possible fixes.

**Comment by nohwnd** [comment]
@dlwyatt Any fix I can come up with, pollutes the assertion just to solve this edge case that did not surface for years. I suggest to document this problem and suggest to escape the path with two back-ticks like this: ````"TestDrive:\``[test``].txt"````. I will also add tests for the function and env drives. What do you think? Any better ideas?

File: Functions/Assertions/Exist.ps1 [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
 function PesterExist($value) {
-    return (Test-Path $value)
+    $resolvedPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($value)
```
Comment (dlwyatt) [inline Functions/Assertions/Exist.ps1]: This line is going to break things (because it strips off the provider name from PS paths), and isn't necessary anyway.  Just the call to `Test-Path -LiteralPath` should be sufficient, I think.

Comment (nohwnd) [inline Functions/Assertions/Exist.ps1]: Yup, I realized that as well after the test failed on CI. How about just documenting it, adding more tests and keeping the assertion as it? https://github.com/pester/Pester/pull/348#issuecomment-101235064

**Comment by dlwyatt** [comment]
Honestly, I think the value of `Should Exist` is way down now that our failed assertions show the content of the line in the test code.  `Test-Path -LiteralPath $something | Should Be $true` is now a perfectly acceptable test, where before I would have preferred `Should Exist`.  So I wouldn't bother with any convoluted escaping of paths in the workaround or documentation.  :)

**Comment by nohwnd** [comment]
Good point. I will document that, including your "work around" and close the issue.
